scipy/_lib/_array_api.py

```diff
@@ -283,13 +283,20 @@ def xp_assert_equal(actual, desired, check_namespace=True, check_dtype=True,
     return np.testing.assert_array_equal(actual, desired, err_msg=err_msg)
 
 
-def xp_assert_close(actual, desired, rtol=1e-07, atol=0, check_namespace=True,
+def xp_assert_close(actual, desired, rtol=None, atol=0, check_namespace=True,
                     check_dtype=True, check_shape=True, err_msg='', xp=None):
     __tracebackhide__ = True  # Hide traceback for py.test
     if xp is None:
         xp = array_namespace(actual)
     desired = _strict_check(actual, desired, xp, check_namespace=check_namespace,
                             check_dtype=check_dtype, check_shape=check_shape)
+    
+    floating = xp.isdtype(desired.dtype, ('real floating', 'complex floating'))
+    if rtol is None and floating:
+        rtol = xp.finfo(desired.dtype).eps**0.5 * 4
```

Comment:
I didn't think about this carefully before, and I had it like this, but I started wondering whether it should be:
```suggestion
    floating = xp.isdtype(actual.dtype, ('real floating', 'complex floating'))
    if rtol is None and floating:
        rtol = xp.finfo(actual.dtype).eps**0.5 * 4
```
If the developer decides to turn off the dtype check, what do they expect to happen when the dtypes are different / what would be most useful?

Suppose `rtol` is based on the reference dtype (as it is now).
- If we are comparing a low-precision actual value against a high-precision reference, that is too hard of a test to pass.
- If we are comparing a high-precision actual value against a low-precision reference, that is too easy of a test to pass.

So perhaps this should be the other way around - the actual value should be compared against the reference value according to the actual value's own precision.